java-client/src/main/java/org/opensearch/client/util/PathEncoder.java

```diff
@@ -8,59 +8,37 @@
 
 package org.opensearch.client.util;
 
-/*
- * SPDX-License-Identifier: Apache-2.0
- *
- * The OpenSearch Contributors require contributions made to
- * this file be licensed under the Apache-2.0 license or a
- * compatible open source license.
- */
-
-import java.lang.invoke.MethodHandle;
-import java.lang.invoke.MethodHandles;
-import java.lang.invoke.MethodType;
-import java.nio.charset.Charset;
-import java.nio.charset.StandardCharsets;
-import java.util.Collections;
-
 public class PathEncoder {
-    private final static String HTTP_CLIENT4_UTILS_CLASS = "org.apache.http.client.utils.URLEncodedUtils";
-    private final static String HTTP_CLIENT5_UTILS_CLASS = "org.apache.hc.core5.net.URLEncodedUtils";
-    private final static MethodHandle FORMAT_SEGMENTS_MH;
-
-    static {
-        Class<?> clazz = null;
-        try {
-            // Try Apache HttpClient5 first since this is a default one
-            clazz = Class.forName(HTTP_CLIENT5_UTILS_CLASS);
-        } catch (final ClassNotFoundException ex) {
-            try {
-                // Fallback to Apache HttpClient4
-                clazz = Class.forName(HTTP_CLIENT4_UTILS_CLASS);
-            } catch (final ClassNotFoundException ex1) {
-                clazz = null;
-            }
+    /**
+     * Percent encoding codec that matches Apache HTTP Client 4's path segment encoding.
+     */
+    @Deprecated
+    public static final PercentCodec APACHE_HTTP_CLIENT_4_EQUIV_CODEC = PercentCodec.RFC3986_PATHSAFE;
+    /**
+     * Percent encoding codec that matches Apache HTTP Client 5's path segment encoding.
+     */
+    public static final PercentCodec APACHE_HTTP_CLIENT_5_EQUIV_CODEC = PercentCodec.RFC3986_UNRESERVED;
+
+    public static final PercentCodec DEFAULT_CODEC = APACHE_HTTP_CLIENT_5_EQUIV_CODEC;
+
+    private static PercentCodec codec;
```

Comment:
Since we leave an option to change the value from anywhere:

```suggestion
    private static volatile PercentCodec codec;
```